Instance reports an error, after another instance closed.

实际场景在本地和K8S集群中都可以重现，比如k8S集群缩容时。
这是因为网络问题导致的？


2023-04-28T10:33:15.086539482+08:00 2023-04-28 10:33:15.086 [,] [pool-7-thread-5] ERROR o.apache.rocketmq.streams.core.state.RocketMQStore - [lambda$removeState$1,279] - remove state error

2023-04-28T10:33:15.086594456+08:00 java.lang.NullPointerException: null

2023-04-28T10:33:15.086601395+08:00 	at java.util.AbstractCollection.addAll(AbstractCollection.java:343)

2023-04-28T10:33:15.086605769+08:00 	at org.apache.rocketmq.streams.core.state.AbstractStore$Wrapper.getAll(AbstractStore.java:156)

2023-04-28T10:33:15.086609858+08:00 	at org.apache.rocketmq.streams.core.state.AbstractStore.getAll(AbstractStore.java:53)

2023-04-28T10:33:15.086614229+08:00 	at org.apache.rocketmq.streams.core.state.RocketMQStore.lambda$removeState$1(RocketMQStore.java:267)

2023-04-28T10:33:15.086618636+08:00 	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)

2023-04-28T10:33:15.086641504+08:00 	at java.util.concurrent.FutureTask.run(FutureTask.java:266)

2023-04-28T10:33:15.086645641+08:00 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)

2023-04-28T10:33:15.086649929+08:00 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)

2023-04-28T10:33:15.086653477+08:00 	at java.lang.Thread.run(Thread.java:748)

2023-04-28T10:33:15.091102054+08:00 2023-04-28 10:33:15.090 [,] [RebalanceService] ERROR o.a.rocketmq.streams.core.running.WorkerThread - [lambda$new$0,145] - recover error.

2023-04-28T10:33:15.091135289+08:00 java.util.concurrent.ExecutionException: java.lang.RuntimeException: java.lang.NullPointerException

2023-04-28T10:33:15.091142105+08:00 	at java.util.concurrent.FutureTask.report(FutureTask.java:122)

2023-04-28T10:33:15.091146330+08:00 	at java.util.concurrent.FutureTask.get(FutureTask.java:206)

2023-04-28T10:33:15.091150340+08:00 	at org.apache.rocketmq.streams.core.state.RocketMQStore.removeState(RocketMQStore.java:285)

2023-04-28T10:33:15.091155282+08:00 	at org.apache.rocketmq.streams.core.state.RocketMQStore.recover(RocketMQStore.java:85)

2023-04-28T10:33:15.091159248+08:00 	at org.apache.rocketmq.streams.core.running.WorkerThread$PlanetaryEngine.lambda$new$0(WorkerThread.java:142)

2023-04-28T10:33:15.091162330+08:00 	at org.apache.rocketmq.streams.core.running.MessageQueueListenerWrapper.messageQueueChanged(MessageQueueListenerWrapper.java:64)

2023-04-28T10:33:15.091165820+08:00 	at org.apache.rocketmq.client.impl.consumer.RebalanceLitePullImpl.messageQueueChanged(RebalanceLitePullImpl.java:53)

2023-04-28T10:33:15.091168893+08:00 	at org.apache.rocketmq.client.impl.consumer.RebalanceImpl.rebalanceByTopic(RebalanceImpl.java:369)

2023-04-28T10:33:15.091173800+08:00 	at org.apache.rocketmq.client.impl.consumer.RebalanceImpl.doRebalance(RebalanceImpl.java:252)

2023-04-28T10:33:15.091178229+08:00 	at org.apache.rocketmq.client.impl.consumer.DefaultLitePullConsumerImpl.doRebalance(DefaultLitePullConsumerImpl.java:1109)

2023-04-28T10:33:15.091182348+08:00 	at org.apache.rocketmq.client.impl.factory.MQClientInstance.doRebalance(MQClientInstance.java:938)

2023-04-28T10:33:15.091187258+08:00 	at org.apache.rocketmq.client.impl.consumer.RebalanceService.run(RebalanceService.java:41)

2023-04-28T10:33:15.091194086+08:00 	at java.lang.Thread.run(Thread.java:748)

2023-04-28T10:33:15.091200842+08:00 Caused by: java.lang.RuntimeException: java.lang.NullPointerException

2023-04-28T10:33:15.091204858+08:00 	at org.apache.rocketmq.streams.core.state.RocketMQStore.lambda$removeState$1(RocketMQStore.java:280)

2023-04-28T10:33:15.091209289+08:00 	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)

2023-04-28T10:33:15.091214210+08:00 	at java.util.concurrent.FutureTask.run(FutureTask.java:266)

2023-04-28T10:33:15.091217565+08:00 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)

2023-04-28T10:33:15.091220779+08:00 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)

2023-04-28T10:33:15.091224304+08:00 	... 1 common frames omitted

2023-04-28T10:33:15.091227380+08:00 Caused by: java.lang.NullPointerException: null

2023-04-28T10:33:15.091230380+08:00 	at java.util.AbstractCollection.addAll(AbstractCollection.java:343)

2023-04-28T10:33:15.091233731+08:00 	at org.apache.rocketmq.streams.core.state.AbstractStore$Wrapper.getAll(AbstractStore.java:156)

2023-04-28T10:33:15.091236557+08:00 	at org.apache.rocketmq.streams.core.state.AbstractStore.getAll(AbstractStore.java:53)

2023-04-28T10:33:15.091239422+08:00 	at org.apache.rocketmq.streams.core.state.RocketMQStore.lambda$removeState$1(RocketMQStore.java:267)

2023-04-28T10:33:15.091242203+08:00 	... 5 common frames omitted

--- a/core/src/main/java/org/apache/rocketmq/streams/core/state/AbstractStore.java
+++ b/core/src/main/java/org/apache/rocketmq/streams/core/state/AbstractStore.java
@@ -145,12 +145,12 @@ public abstract class AbstractStore {
         }
 
         public Set<byte[]> getInCalculating(String stateTopicQueue) {
-            return calculating.get(stateTopicQueue);
+            return calculating.getOrDefault(stateTopicQueue, new HashSet<>());
         }
 
         public Set<byte[]> getAll(String stateTopicQueue) {
-            Set<byte[]> calculating = this.calculating.get(stateTopicQueue);
-            Set<byte[]> recover = this.recover.get(stateTopicQueue);
+            Set<byte[]> calculating = this.calculating.getOrDefault(stateTopicQueue, new HashSet<>());
+            Set<byte[]> recover = this.recover.getOrDefault(stateTopicQueue, new HashSet<>());
 
             Set<byte[]> result = new HashSet<>();
             result.addAll(calculating);
@@ -162,7 +162,7 @@ public abstract class AbstractStore {
 
         public String whichStateTopicQueueBelongTo(byte[] key) {
             for (String uniqueQueue : recover.keySet()) {
-                for (byte[] tempKeyByte : recover.get(uniqueQueue)) {
+                for (byte[] tempKeyByte : recover.getOrDefault(uniqueQueue, new HashSet<>())) {
                     if (Arrays.equals(tempKeyByte, key)) {
                         return uniqueQueue;
                     }
@@ -170,7 +170,7 @@ public abstract class AbstractStore {
             }
 
             for (String uniqueQueue : calculating.keySet()) {
-                for (byte[] tempKeyByte : calculating.get(uniqueQueue)) {
+                for (byte[] tempKeyByte : calculating.getOrDefault(uniqueQueue, new HashSet<>())) {
                     if (Arrays.equals(tempKeyByte, key)) {
                         return uniqueQueue;
                     }

